app/src/main/java/org/openimis/imispolicies/ClientAndroidInterface.java

```diff
@@ -3638,15 +3639,20 @@ private int Enrol(int oFamilyId, int oInsureeId, int oPolicyId, int oPremiumId,
                         familyObj.put("insurees", tempInsureesArray);
 
                         // Policy + premium
-                        JSONArray tempPoliciesArray = new JSONArray();
 
                         for (int j = 0; j < policiesArray.length(); j++) {
-                            tempPoliciesArray = policiesArray;
-                            JSONObject premiumObj = premiumsArray.getJSONObject(j);
-                            tempPoliciesArray.getJSONObject(j).put("premium", premiumObj);
+                            JSONArray policyPremiums = new JSONArray();
+                            String policyId = policiesArray.getJSONObject(j).getString("PolicyId");
+                            for(int k = 0; k < premiumsArray.length(); k++) {
+                                JSONObject premiumObject = premiumsArray.getJSONObject(k);
+                                if(StringUtils.equals(policyId,premiumObject.getString("PolicyId"))) {
```

Comment:
whitespace after ",". Please move PolicyID string to constant

Reply:
whitespace after "if"

Reply:
@malinowskikam you forgot about moving PolicyID string to constant

Reply:
@jmadej I forgot to mention this, but every json object access in this project is done with string literals, and fixing this in only one place would just add into the confusion. (for instance this two literals actually mean different fields, and should be in 2 separate constants (one for policy schema and one for premium schema)

Reply:
I see, thanks for explanation